client/layout/index.jsx

```diff
@@ -152,6 +153,8 @@ class Layout extends Component {
 	};
 
 	componentDidMount() {
+		detectGeolocation();
```

Comment:
I'd have suggested doing this in a component like the query components we have, but then calling it as early as possible makes sense too. 

Alternatively, we could do that even earlier at boot time:

https://github.com/Automattic/wp-calypso/blob/6a8c4cc88dad56a331def849f5acea515cc37f53/client/boot/common.js#L421-L450

which would also have a good side benefit of not having to repeat it in both layout components.

Reply:
Ah, nice idea! I've moved the call into the `boot` function. Let me know if that looks right to you.